Introducing service principal creation policies (#13608)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/serviceprincipalcreationpolicy-post-includes.md

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Add conditions under which a service principal creation event is *included* in a service principal creation policy. You do this by adding a [servicePrincipalCreationConditionSet](../resources/serviceprincipalcreationconditionset.md) object to the **includes** collection of a  [servicePrincipalCreationPolicy](../resources/servicePrincipalCreationPolicy.md).

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
| Delegated (work or school account) | Policy.ReadWrite.ServicePrincipalCreate |
|Delegated (personal Microsoft account)| Not supported. |
| Application | Policy.ReadWrite.ServicePrincipalCreate |

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /policies/servicePrincipalCreationPolicies/{servicePrincipalCreationPolicyId}/includes
```

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

> In the request body, supply a JSON representation of the [servicePrincipalCreationConditionSet](../resources/serviceprincipalcreationconditionset.md) object.
> 
> The following table shows the properties that are required when you create the [servicePrincipalCreationConditionSet](../resources/serviceprincipalcreationconditionset.md).

Reviewer: Correcting the wrong stub gen boilerplate text
```suggestion
In the request body, supply a JSON representation of the [servicePrincipalCreationConditionSet](../resources/serviceprincipalcreationconditionset.md) object. You can specify the following properties when creating a [servicePrincipalCreationConditionSet](../resources/serviceprincipalcreationconditionset.md).
```

|Property|Type|Description|
|:---|:---|:---|

> | id | String | The unique identifier for the service principal creation condition set. Key. Read-only. |
> | applicationIds | String collection | A list of **appId** values for the applications to match with, or a list with the single value `all` to match any application. Default is the single value `all`. |
> |applicationPublisherIds | String collection | A list of Microsoft Partner Network (MPN) IDs for verified publishers of the application, or a list with the single value `all` to match with applications from any publisher. Default is the single value `all`. |
> | applicationsFromVerifiedPublisherOnly | Boolean | Set to `true` to only match on applications with a verified publisher. Set to `false` to match on any application, even if it does not have a verified publisher. Default is `false` |
> | applicationTenantIds | String collection | A list of Azure Active Directory tenant IDs in which the application is registered, or a list with the single value `all` to match with applications registered in any tenant. Default is the single value `all`. |

Reviewer: Updating the table to complete the boilerplate text. Removing **id** because it's read-only.
```suggestion
| applicationIds | String collection | A list of **appId** values for the applications to match with, or a list with the single value `all` to match any application. Default is the single value `all`. Optional. |
|applicationPublisherIds | String collection | A list of Microsoft Partner Network (MPN) IDs for verified publishers of the application, or a list with the single value `all` to match with applications from any publisher. Default is the single value `all`. Optional. |
| applicationsFromVerifiedPublisherOnly | Boolean | Set to `true` to only match on applications with a verified publisher. Set to `false` to match on any application, even if it does not have a verified publisher. Default is `false`.  Optional.|
| applicationTenantIds | String collection | A list of Azure Active Directory tenant IDs in which the application is registered, or a list with the single value `all` to match with applications registered in any tenant. Default is the single value `all`. Optional. |
```
